R/ascii_tables.R

```diff
@@ -108,38 +112,87 @@ as_ascii_table <- function(dat, line_prefix = "  ") {
     )
 }
 
+#' Format vector to printable string
+#'
+#' Coerces a vector of any type into a printable string. The most
+#' significant transformation is performed on existing character
+#' vectors which will be truncated, have newlines converted
+#' to explicit symbols and will be wrapped in quotes if they
+#' contain white space.
+#'
+#' @param x (`vector`) \cr vector to be converted to character
+#' @param add_quotes (`logical`) \cr if true will wrap strings that contain
+#' whitespace with quotes
+#' @param crop_at (`numeric`) \cr specifies the limit at which strings should
+#' be truncated to
+#' @param ... additional arguments (not currently used)
+#'
+#' @name as_fmt_char
+#' @keywords internal
+as_fmt_char <- function(x, ...) {
+    UseMethod("as_fmt_char")
+}
 
+#' @rdname as_fmt_char
+#' @export
+as_fmt_char.numeric <- function(x, ...) {
+    format(x, digits = 7, justify = "right")
+}
 
+#' @rdname as_fmt_char
+#' @export
+as_fmt_char.NULL <- function(x, ...) {
+    "<NULL>"
+}
 
+#' @rdname as_fmt_char
+#' @export
+as_fmt_char.list <- function(x, ...) {
+    vapply(
+        x,
+        function(x) {
+            if (is.numeric(x)) {
+                return(as_fmt_char(x))
+            }
+            if (is.character(x) & length(x) == 1) {
+                return(as_fmt_char(x))
+            }
+            as_fmt_char(
+                paste(capture.output(dput(x)), collapse = " "),
```

Comment:
a thought, we could use https://cynkra.github.io/constructive/ which I saw at UseR. The main reason this occurs to me is the odd look of the iris dataset. Perhaps unecessary as we maybe just should expect a printed data frame to look weird, but might be worth trying?